autogluon/task/tabular_prediction/predictor.py

```diff
@@ -624,6 +627,77 @@ def info(self):
         """
         return self._learner.get_info(include_model_info=True)
 
+    # TODO: Handle cases where name is same as a previously fit model, currently overwrites old model.
+    # TODO: Add dataset argument
+    # TODO: Add option to disable OOF generation of newly fitted models
+    # TODO: Move code logic to learner/trainer
+    # TODO: Add task.fit arg to perform this automatically at end of training
+    def fit_weighted_ensemble(self, base_models: list = None, name_suffix='_custom', expand_pareto_frontier=False, time_limits=None):
+        """
+        [EXPERIMENTAL] Fits new weighted ensemble models.
+        Warning: This functionality is currently in preview mode.
+            The API may change entirely in future versions without warning.
+
+        Parameters
+        ----------
+        base_models : list, default = None
+            List of model names the weighted ensemble can use.
+            If None, all previously trained models are used except for weighted ensemble models.
+            As an example, to train a weighted ensemble that can only have weights assigned to the models 'model_a' and 'model_b', set `base_models=['model_a', 'model_b']`
+        name_suffix : str, default = '_custom'
+            Name suffix to add to the name of the newly fitted model.
+            Warning: If the name of a trained model from this function is identical to an existing model, it will overwrite the existing model.
+            Ensure that `name_suffix` is unique each time `fit_weighted_ensemble` is called to avoid this.
+        expand_pareto_frontier : bool, default = False
+            If True, will train N-1 weighted ensemble models instead of 1, where `N=len(base_models)`.
+            The final model trained when True is equivalent to the model trained when False.
+            These weighted ensemble models will attempt to expand the pareto frontier.
+            This is useful when inference speed is an important consideration.
```

Comment:
This will create many different weighted ensembles which have different accuracy/memory/inference-speed trade-offs.

Reply:
Added